cfme/tests/cloud_infra_common/test_genealogy.py

```diff
@@ -87,6 +88,10 @@ def _():
             parent = pytest.sel.text(opt).strip()
         assert parent.startswith(small_template), "The parent template not detected!"
     else:
-        vm_crud_ancestors = vm_crud.genealogy.ancestors
+        try:
+            vm_crud_ancestors = vm_crud.genealogy.ancestors
+        except NameError:
+            logger.exception("The parent template not detected!")
+            raise pytest.fail("The parent template not detected!")
```

Comment:
This is a parent template/vm of a current vm, in genealogy context. If NameError occurs, no parent has been detected. The error message displayed reflects this, I believe.